### ISSUE TITLE
Memory leak with SWIG wrapper

### ISSUE THREAD

**Issue by jsdessureault** [cmusphinx/pocketsphinx#51]
Hi, 

I'm having a tricky bug when I try to create my model (pocketsphinx + SWIG wrapper for Python, under Ubuntu Trustee Tahr, running on a Raspberry PI 2).  

```
config = Decoder.default_config()
config.set_string('-hmm', hmm)
```

This ".set_string" gives me this message: 

> swig/python detected a memory leak of type 'Config *', no destructor found.

I really don't know what to change in the wrapper to fix it.  

Any idea? 

In advance, thanks! 

JS

**Comment by nshmyrev** [comment]
What is your swig version?

**Comment by jsdessureault** [comment]
That's 2.0.11

**Comment by jsdessureault** [comment]
Obviously, my hmm parameter has been loaded correctly... This is not the problem.

**Comment by nshmyrev** [comment]
Something is wrong with the code you compiled. What is your python version. What is your pocketsphinx version, what is your sphinxbase version, how exactly you compiled it, what exactly is the code you run.

**Comment by nshmyrev** [comment]
It is better to provide build logs for pocketsphinx

**Comment by jsdessureault** [comment]
Ok, here's the information:
- Python version: 2.7.6
- Pocketsphinx version: 5prealpha
- Sphinxbase version: 5prealpha.

I've compiled both with "sh autogen.sh" and then "make". 

In pocketsphinx "make", there was an interesting warning: 

> .../swig/ps_decoder.i:54: Warning 504: Function Decoder::%Decoder(Config *, int *) must have a return type.  Ignored. 

Code here: 
https://github.com/jsdessureault/spike/blob/master/behavior_ecoute.py 
lines 42-43:

> swig/python detected a memory leak of type 'Config *', no destructor found.

I can't see why,,,

**Comment by nshmyrev** [comment]
I can not reproduce your situation, even your warning. I strongly suspect you have some different sphinxbase version, probably an old one.

Please provide pocketsphinx build log

Please provide md5sums of `.i` files in `sphinxbase/swig` and in `pocketsphinx/swig`

**Comment by jsdessureault** [comment]
Ok, I have uninstalled/reinstalled everything from scratch. 
I had the same error message.   But I've tried to execute within a ROS node (it was the goal, anyway), I it seems to be way better.   Not working yet, be it appears to be a parameter problem between the launcher and the node (ros-jade-pocketsphinx).   I think I'll be able to fix it next time I work on it.  
Thank you very much for your help!!
